### ISSUE TITLE
Add option to disable tests

### ISSUE THREAD

**Issue by nurupo** [AppImage/AppImageKit#788]
`cmake -DBUILD_TESTING=OFF` doesn't work. Cmake still requires `GTest` to be present on the system, and if it's not, it will download and build it despite `BUILD_TESTING=OFF`. It also builds the tests themselves with `BUILD_TESTING=OFF`.
```
$ cmake -DBUILD_TESTING=OFF ..
...
-- Downloading and building GTest
...
Scanning dependencies of target test-xdg-basedir
[ 97%] Building CXX object src/xdg-basedir/CMakeFiles/test-xdg-basedir.dir/test-xdg-basedir.cpp.o
[100%] Linking CXX executable test-xdg-basedir
```
I'm pretty sure that this option does nothing, so we need a new and better `DISABLE_TESTS` or something.

**Comment by TheAssassin** [comment]
Interesting find. Actually, it is used by us here: https://github.com/AppImage/AppImageKit/blob/appimagetool/master/tests/CMakeLists.txt

**Comment by nurupo** [comment]
Yes, I know that it's used there, but I seems to always evaluate to true, perhaps because it's not cached. Use [`opton()`](https://cmake.org/cmake/help/v3.5/command/option.html) for ON and OFF and cached strings for everything else.

E.g.

```cmake
option(ENABLE_TESTS "Wether to enable tests and require GTest" OFF)
if (ENABLE_TESTS)
  ...
endif()
    

set(SOME_STRING_VAR "The default value if no specified, can be empty string" CACHE STRING "Description of what this var does")
```

**Comment by TheAssassin** [comment]
> The module automatically creates a BUILD_TESTING option that selects whether to enable testing support (ON by default). After including the module, use code like:
>
> ```
> if(BUILD_TESTING)
>   # ... CMake code to create tests ...
> endif()
> ```

https://cmake.org/cmake/help/v3.7/module/CTest.html

**Comment by TheAssassin** [comment]
Also, `set()` can be used for boolean values. Just call like `set(something ON CACHE BOOL "")`.

**Comment by probonopd** [comment]
For the "rest of us" who are not CMake experts, can BUILD_TESTING be set as a command line option without the need to change anything inside a CMake file?

**Comment by TheAssassin** [comment]
@probonopd that's the idea, but it doesn't work according to @nurupo.

**Comment by probonopd** [comment]
right... :)

**Comment by TheAssassin** [comment]
Fixed in #807.
